crates/biome_css_parser/src/syntax/value/type.rs

```diff
@@ -0,0 +1,292 @@
+use biome_parser::parse_recovery::ParseRecovery;
+use biome_parser::parse_recovery::RecoveryResult;
+use biome_parser::parsed_syntax::ParsedSyntax::{Absent, Present};
+use biome_parser::token_set;
+use biome_parser::{Parser, prelude::ParsedSyntax};
+
+use crate::parser::CssParser;
+use crate::syntax::is_at_identifier;
+use crate::syntax::is_at_string;
+use crate::syntax::parse_regular_identifier;
+use crate::syntax::parse_string;
+use crate::syntax::value::parse_error::expected_syntax_component;
+
+const SYNTAX_MULTIPLIER_SET: TokenSet<CssSyntaxKind> = token_set![T![#], T![+]];
+
+const KNOWN_SYNTAX_TYPE_NAMES: [&str; 14] = [
+    "angle",
+    "color",
+    "custom-ident",
+    "image",
+    "integer",
+    "length",
+    "length-percentage",
+    "number",
+    "percentage",
+    "resolution",
+    "string",
+    "time",
+    "url",
+    "transform-function",
+];
+
+const SYNTAX_KIND_WITHOUT_MULTIPLIER: [&str; 1] = ["transform-list"];
+
+#[inline]
+pub(crate) fn is_at_type_function(p: &mut CssParser) -> bool {
+    p.at(T![type]) && p.nth_at(1, T!['('])
+}
+
+/// Parses a type function from the current position of the CSS parser.
+/// For more detailed information on the CSS type function syntax, refer to the [CSS Values and
+/// Units Module](https://drafts.csswg.org/css-values-5/#typedef-syntax)
+///
+/// # Type Function Syntax Examples
+///
+/// - Single value:
+///   ``` css
+///   type(<color>)
+///   type(auto)
+///   ```
+/// - "|" combinator for multiple types:
+///   ``` css
+///   type(<length> | <percentage>)
+///   ```
+/// - Comma-separated list of values
+///   ```css
+///   type(<color>+)
+///   ```
+/// - Comma-separated list of values
+///   ```css
+///   type(<length>#)
+///   ```
+/// - Multiple keywords
+///   ```css
+///   type(red | blue | green)
+///   ```
+/// - Combination of data type and keyword
+///   ```css
+///   type(<percentage> | auto)
+///   ```
+/// - Universal syntax value
+///   ```css
+///   type(*)
+///   ```
+///
+/// # Grammar
+///
+/// ``` txt
+/// type( <syntax> )
+///
+/// <syntax> = '*' | <syntax-component> [ <syntax-combinator> <syntax-component> ]* | <syntax-string>
+/// <syntax-component> = <syntax-single-component> <syntax-multiplier>?
+///                    | '<' transform-list '>'
+/// <syntax-single-component> = '<' <syntax-type-name> '>' | <ident>
+/// <syntax-type-name> = angle | color | custom-ident | image | integer
+///                    | length | length-percentage | number
+///                    | percentage | resolution | string | time
+///                    | url | transform-function
+/// <syntax-combinator> = '|'
+/// <syntax-multiplier> = [ '#' | '+' ]
+///
+/// <syntax-string> = <string>
+#[inline]
+pub(crate) fn parse_type_function(p: &mut CssParser) -> ParsedSyntax {
+    if !is_at_type_function(p) {
+        return Absent;
+    }
+
+    let m = p.start();
+
+    p.bump(T![type]);
+    p.bump(T!['(']);
+    parse_any_syntax(p).ok();
```

Comment:
Shouldn't we raise an error diagnostic if there isn't any `<syntax>`?

Reply:
Added [this](https://github.com/biomejs/biome/pull/8255/files#diff-b163ef330ad3a8eba29e2382fd375d270bef74146b67b19bef573083c388920aR112-R114), let me know if that is the right pattern or not